src/main/java/de/tum/cit/aet/artemis/core/service/ArtemisSuccessfulLoginService.java

```diff
@@ -35,11 +36,11 @@ public class ArtemisSuccessfulLoginService {
 
     private static final Logger log = LoggerFactory.getLogger(ArtemisSuccessfulLoginService.class);
 
-    @Value("${artemis.user-management.password-reset.links.en}")
-    private String passwordResetLinkEnUrl;
+    @Value("${artemis.user-management.password-reset.links.en:#{null}}")
+    private Optional<String> passwordResetLinkEnUrl;
```

Comment:
When using an optional we need to unwrap it before using it in the mail template (119 & 122)

Reply:
Would it make sense to define another variable for the reset link that is not optional then? After the PostConstruct the value cannot be undefined anymore, so we do not need an optional for the value used in the template, but might want to keep the variable that reads the URL from the config optional for consistency

Reply:
Resolved by using constructor-based injection for the values instead.